addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1348,27 +1348,26 @@ const Wysiwyg = Widget.extend({
      * @param {Node} [params.htmlClass] Optionnal
      */
     openMediaDialog(params = {}) {
-        const sel = this.odooEditor.document.getSelection();
-
-        if (!sel.rangeCount) {
-            return;
-        }
-        const range = sel.getRangeAt(0);
         // We lose the current selection inside the content editable when we
         // click the media dialog button so we need to be able to restore the
         // selection when the modal is closed.
         const restoreSelection = preserveCursor(this.odooEditor.document);
 
-        const $editable = $(OdooEditorLib.closestElement(range.startContainer, '.o_editable') || this.odooEditor.editable);
-        const model = $editable.data('oe-model');
-        const field = $editable.data('oe-field');
-        const type = $editable.data('oe-type');
+        const $el = $(params.node || this.odooEditor.document.getSelection().anchorNode);
```

Comment:
You changed `range.startContainer` for `selection.anchorNode`. Just FYI, technically that is not equivalent because the `startContainer` is the start of the range in document DFS traversal, whereas the `anchorNode` is the start of the selection from a user point of view, where the user clicked first. If the user made a _backward_ selection, those nodes can be different. That being said, in the case of `openMediaDialog`, I suppose it's going to be the same since the selection is probably collapsed anyway. Depending on what you do about the previous comment (put back the previous code or not?), I'd prefer to switch back to `range.startContainer` so that the behavior is agnostic of the direction in which the user made the selection.

Note that this line can crash just like `preserveCursor` above, as `getSelection()` can return `null`. Again, maybe it is never `null` in the specific context in which this function is called but since we're playing with iframes, especially in website where the sidebar is outside of the iframe, we end up having two documents, each with their own selection, so I feel like not checking the selection is a little like playing with fire.

Reply:
I just check that both return the same element for all the usecas of `openMediaDialog()` but I don't know much about DOM selection so I went back to `startContainer` this way we are more correct :)